app/controllers/merchants_controller.rb

```diff
@@ -6,7 +6,12 @@ def index
   end
 
   def show
-    @merchant = Merchant.find(params[:id])
+    if site_admin?
```

Comment:
Same with my comment above regarding `Admin::BaseController`. If we inherit from it, we already know the user is the site admin, and thus we shouldn't need this if statement.